File: ruby/v2.1/create_advertiser.rb
#!/usr/bin/env ruby
# Encoding: utf-8
#
# Copyright:: Copyright 2015, Google Inc. All Rights Reserved.
#
# License:: Licensed under the Apache License, Version 2.0 (the "License");
#           you may not use this file except in compliance with the License.
#           You may obtain a copy of the License at
#
#           http://www.apache.org/licenses/LICENSE-2.0
#
#           Unless required by applicable law or agreed to in writing, software
#           distributed under the License is distributed on an "AS IS" BASIS,
#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
#           implied.
#           See the License for the specific language governing permissions and
#           limitations under the License.
#
# This example creates an advertiser in a given DCM account.

require_relative 'dfareporting_utils'

def create_advertiser(dfareporting, args)
  # Create a new advertiser resource to insert
  advertiser_group = {
    :name => 'Example Advertiser',
    :status => 'APPROVED'
  }

  # Insert the advertiser
  result = dfareporting.advertisers.insert(
    :profileId => args[:profile_id]
  ).body(advertiser_group).execute()

  # Display results
  puts 'Created advertiser with ID %d and name "%s".' %
      [result.data.id, result.data.name]
end

if __FILE__ == $0
  # Retrieve command line arguments
  args = DfaReportingUtils.get_arguments(ARGV, :profile_id)

  # Authenticate and initialize API service
  dfareporting = DfaReportingUtils.setup()

  create_advertiser(dfareporting, args)
end

File: java/v4/src/main/java/com/google/api/services/samples/dfareporting/advertisers/CreateAdvertiser.java
// Copyright 2014 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.google.api.services.samples.dfareporting.advertisers;

import com.google.api.services.dfareporting.Dfareporting;
import com.google.api.services.dfareporting.model.Advertiser;
import com.google.api.services.samples.dfareporting.DfaReportingFactory;


/**
 * This example creates an advertiser in a given DFA account. To get advertisers, see
 * GetAdvertisers.java.
 */
public class CreateAdvertiser {
  private static final String USER_PROFILE_ID = "INSERT_USER_PROFILE_ID_HERE";

  private static final String ADVERTISER_NAME = "INSERT_ADVERTISER_NAME_HERE";

  public static void runExample(Dfareporting reporting, long profileId, String advertiserName)
      throws Exception {
    // Create the advertiser structure.
    Advertiser advertiser = new Advertiser();
    advertiser.setName(advertiserName);
    advertiser.setStatus("APPROVED");

    // Create the advertiser.
    Advertiser result = reporting.advertisers().insert(profileId, advertiser).execute();

    // Display the advertiser ID.
    System.out.printf("Advertiser with ID %d was created.%n", result.getId());
  }

  public static void main(String[] args) throws Exception {
    Dfareporting reporting = DfaReportingFactory.getInstance();

    long profileId = Long.parseLong(USER_PROFILE_ID);

    runExample(reporting, profileId, ADVERTISER_NAME);
  }
}
